# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/cairo/cairo/src/cairo-quartz-surface.c
# Commit: 61ef06598b0c
# Full Hash: 61ef06598b0c87016b6eecabf22cde2b7f1fbeb6
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2024-05-07 09:13:07
# Regressor Bug: 1892913
# File Overlap Count: 2
# Description:
#   Bug 1892913 - patch 18 - Cairo fixes for iOS build. r=gfx-reviewers,lsalzman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D209426
# ==============================================================================

diff -r 51b400f880e3 -r 61ef06598b0c gfx/cairo/cairo/src/cairo-quartz-surface.c
--- a/gfx/cairo/cairo/src/cairo-quartz-surface.c	Mon May 06 19:25:20 2024 +0000
+++ b/gfx/cairo/cairo/src/cairo-quartz-surface.c	Mon May 06 19:25:20 2024 +0000
@@ -201,8 +201,10 @@
         if (color_space)
             return color_space;
     }
+#if CAIRO_HAS_QUARTZ_APPLICATION_SERVICES /* available on macOS but not iOS */
     if (!color_space)
         color_space =  CGDisplayCopyColorSpace (CGMainDisplayID ());
+#endif
 
     if (!color_space)
         color_space = CGColorSpaceCreateDeviceRGB ();